desc/objectives/objective_funs.py

```diff
@@ -723,9 +724,11 @@ def bounds_scaled(self):
             else:
                 lb_i = jnp.ones(obj.dim_f) * obj.target
                 ub_i = jnp.ones(obj.dim_f) * obj.target
-            if obj._normalize_target:
-                lb_i /= obj.normalization
-                ub_i /= obj.normalization
+            lb_i = obj._scale(lb_i)
+            ub_i = obj._scale(ub_i)
+            if not obj._normalize_target:
+                lb_i *= obj.normalization
+                ub_i *= obj.normalization
```

Comment:
From the linked issue it's clear that it is undesirable to multiply the objective by the quadrature weighs because the target and bounds for the objective were unaware of this scaling. So the solution used here is to also scale the target and bounds by the quadrature weights through the call to `obj._scale`.

`obj._scale` first calls [this method](https://github.com/PlasmaControl/DESC/blob/9428bdaab2ddf50351131b72d258fab362e92828/desc/objectives/_generic.py#L670) for the rotational transform objective in particular (applying the quadrature weight `w`), then in general calls this [method](https://github.com/PlasmaControl/DESC/blob/a2fe711ffa3f3de459ba65e2d5520a91f98841a4/desc/objectives/objective_funs.py#L986) which applies objective weighting.

The changes in `test_objective_funs.py` of multiplying the target and bounds by the objective weights are there now to account for this change. It looks like the target/bounds were previously also unaware of the objective weights which could cause a similar issue to the one discussed in the linked issue.